libraries/azurerm_virtual_network.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'azurerm_resource'
+
+class AzurermVirtualNetwork < AzurermSingularResource
+  name 'azurerm_virtual_network'
+  desc 'Verifies settings for an Azure Virtual Network'
+  example <<-EXAMPLE
+    describe azurerm_virtual_network(resource_group: 'example', name: 'vnet-name') do
+      it { should have_monitoring_agent_installed }
+    end
+  EXAMPLE
+
+  ATTRS = %i(
+    id
+    name
+    location
+    type
+    tags
+    properties
+  ).freeze
+
+  attr_reader(*ATTRS)
+
+  def initialize(resource_group: nil, name: nil)
+ 
+    resp = client.virtual_network(resource_group, name)
+    return if resp.nil? || resp.key?('error')
+
+    ATTRS.each do |field|
+      instance_variable_set("@#{field}", resp[field.to_s])
+    end
+    
+    @subs = begin
+             resp['properties']['subnets']
+            rescue
+                 []
+            end
+    @exists = true
+  end
+
+  def address_space
+    properties['addressSpace']['addressPrefixes']
+  end
+
+  def dns_servers
+      properties['dhcpOptions']['dnsServers']
+  end
+      
+  def vnet_peerings
+    properties['virtualNetworkPeerings'].each_with_object({}) do |peer, hash|
+        hash[ peer['name'] ] = peer['properties']['remoteVirtualNetwork']['id']
+    end
+  end
+
+  def enable_ddos_protection
+      properties['enableDdosProtection']
+  end
+
+  def enable_vm_protection
+      properties['enableVmProtection']
+  end
+
+  def subnets
+    @subs.map do |s|
+        s['name']
+    end
```

Comment:
Prefer the one line with braces unless you hit the line length limit or it's otherwise more clear, please. =^)
```ruby
@subs.collect { |s| s['name'] }
```